includes/User.php

```diff
@@ -2740,7 +2740,12 @@ private function shouldGetAttributeFromService( $attributeName ) {
 	 * @see getGlobalAttribute for more documentation about attributes
 	 */
 	public function setGlobalAttribute($attribute, $value) {
-		$this->setOptionHelper($attribute, $value);
+		if ( $this->isPublicAttribute( $attribute ) ) {
+			$value = $this->replaceNewlineAndCRWithSpace( $value );
+			$this->userAttributes()->setAttribute( $this->getId(), new Attribute( $attribute, $value ) );
+		}
+
+		$this->setOptionHelper( $attribute, $value );
```

Comment:
Continue to set global attributes in MW for the time being. This is necessary for the logging I added to try and identify conditions where MW and the Service are having different values saved for the same attribute.

When we do SOC-1408 (remove public attributes from the MW dbs) we'll make sure to stop setting them here.